x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/action/PutConnectorAction.java

```diff
@@ -0,0 +1,253 @@
+
+        public Request(
+            String connectorId,
+            String description,
+            String indexName,
+            Boolean isNative,
+            String language,
+            String name,
+            String serviceType
+        ) {
+            this.connectorId = connectorId;
+            this.description = description;
+            this.indexName = indexName;
+            this.isNative = isNative;
+            this.language = language;
+            this.name = name;
+            this.serviceType = serviceType;
+        }
+
+        public Request(StreamInput in) throws IOException {
+            super(in);
+            this.connectorId = in.readString();
+            this.description = in.readOptionalString();
+            this.indexName = in.readOptionalString();
+            this.isNative = in.readOptionalBoolean();
+            this.language = in.readOptionalString();
+            this.name = in.readOptionalString();
+            this.serviceType = in.readOptionalString();
+        }
+
+        private static final ConstructingObjectParser<Request, String> PARSER = new ConstructingObjectParser<>(
+            "connector_put_request",
+            false,
+            ((args, connectorId) -> new Request(
+                connectorId,
+                (String) args[0],
+                (String) args[1],
+                (Boolean) args[2],
+                (String) args[3],
+                (String) args[4],
+                (String) args[5]
+            ))
+        );
+
+        static {
+            PARSER.declareString(optionalConstructorArg(), new ParseField("description"));
+            PARSER.declareString(optionalConstructorArg(), new ParseField("index_name"));
+            PARSER.declareBoolean(optionalConstructorArg(), new ParseField("is_native"));
+            PARSER.declareString(optionalConstructorArg(), new ParseField("language"));
+            PARSER.declareString(optionalConstructorArg(), new ParseField("name"));
+            PARSER.declareString(optionalConstructorArg(), new ParseField("service_type"));
+        }
+
+        public static Request fromXContentBytes(String connectorId, BytesReference source, XContentType xContentType) {
+            try (XContentParser parser = XContentHelper.createParser(XContentParserConfiguration.EMPTY, source, xContentType)) {
+                return Request.fromXContent(parser, connectorId);
+            } catch (IOException e) {
+                throw new ElasticsearchParseException("Failed to parse: " + source.utf8ToString(), e);
+            }
+        }
+
+        public static Request fromXContent(XContentParser parser, String connectorId) throws IOException {
+            return PARSER.parse(parser, connectorId);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.field("description", description);
```

Comment:
Question, why are we not encoding the connector id into xcontent here?

Reply:
The `AbstractBWCSerializationTestCase` requires the `PutConnectorAction.Request` to implement the toXContent method. I didn't include the `connectorId` as it's not parsed directly from JSON body. 

It seems that toXContent is only used for BWC tests, it's not used in the codebase for anything else